packages/core/strapi/lib/middlewares/router/index.js

```diff
@@ -41,15 +41,33 @@ module.exports = strapi => {
   };
 
   const registerAPIRoutes = () => {
-    strapi.router.prefix(strapi.config.get('middleware.settings.router.prefix', ''));
+    for (const apiName in strapi.api) {
+      const api = strapi.api[apiName];
 
-    for (const route of strapi.config.routes) {
-      composeEndpoint(route, { router: strapi.router });
+      _.forEach(api.routes, route => {
+        // nested router
+        if (_.has(route, 'routes')) {
+          for (const key in route.routes) {
+            const routerInfo = route.routes[key];
+
+            const router = new Router({ prefix: routerInfo.prefix });
+
+            for (const route of routerInfo.routes || []) {
+              composeEndpoint(route, { apiName, router });
+            }
+
+            strapi.router.use(router.routes()).use(router.allowedMethods());
+          }
```

Comment:
If I understood well, the structure is as follow ?
```js
{
  country: { // api
    routes: [{ // _.forEach(api.routes
      routes: { // for (const key in route.routes
        myRouter: {
          routes: [{...}], // for (const route of routerInfo.routes
        }
      },
    }]
  }
}
```

Reply:
```json
{
  apiName: {
    routes: {
      country: {
        routes: {
          prefix: '',
          routes: [{ ... }]
        }
      }
    }
  }
}
```